ovirt/resource_ovirt_vm.go

```diff
@@ -376,6 +382,71 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	vmBuilder.Cpu(cpu)
 
+	if storage_domain, _ := blockDevice.([]interface{})[0].(map[string]interface{})["storage_domain"]; storage_domain != "" && templateIDOK {
+
+		// Get the reference to the service that manages the storage domains
+		sdsService := conn.SystemService().StorageDomainsService()
+
+		// Find the storage domain we want to be used for virtual machine disks
+		sdsResp, err := sdsService.List().Search("name=" + storage_domain.(string)).Send()
+		if err != nil {
+			return fmt.Errorf("Failed to search storage domains, reason: %v", err)
+		}
+
+		tds, err := getTemplateDiskAttachments(templateID.(string), meta)
+		if err != nil {
+			return fmt.Errorf("Failed to get Template disks attachments: %v", err)
+		}
+
+		if sdSlice, ok := sdsResp.StorageDomains(); ok {
+			sd := sdSlice.Slice()[0]
+
+			for i, v := range tds {
+				diskIndex := i + 1
+				disk := v.MustDisk()
+				disk.SetStorageDomain(sd)
+
+				// Define basic disk aliases only if attribute alias defined
+				if alias, _ := blockDevice.([]interface{})[0].(map[string]interface{})["alias"]; alias != "" {
+					_, diskBotable := disk.Bootable()
+					switch diskBotable {
+					case true:
+						disk.SetAlias(alias.(string))
+					case false:
+						disk.SetAlias(fmt.Sprintf("%s_Disk%v", d.Get("name").(string), diskIndex))
+					}
+					diskattachment := ovirtsdk4.NewDiskAttachmentBuilder().
+						Disk(ovirtsdk4.NewDiskBuilder().
+							Id(disk.MustId()).
+							Format(ovirtsdk4.DISKFORMAT_COW).
+							Alias(disk.MustAlias()).
+							StorageDomainsOfAny(
+								ovirtsdk4.NewStorageDomainBuilder().
+									Id(sd.MustId()).
+									MustBuild()).
+							MustBuild()).
+						MustBuild()
+
+					vmBuilder.DiskAttachmentsOfAny(diskattachment)
+
+				} else {
```

Comment:
I think what you wanted is roughly this (writing in pseudo-code):
```
diskattachmentBuilder := ovirtsdk4.NewDiskAttachmentBuilder().
						Disk(ovirtsdk4.NewDiskBuilder().
							... whatever needed...
if aliasExists {
         diskattachmentBuilder.Alias("the alias you wanted")
}

Reply:
I tried to use this condition when alias is not defined
```
if aliasExists {
      diskattachmentBuilder.Alias("")
}
```
Any value inside diskattachmentBuilder.Alias(value) cause override alias, specified for disk inside Template.
I don't want override disk alias from Template if the customer didn't configure the new Alias in the terraform.
Therefore created a condition which creates diskattachment with Alias parameter (when customer configured alias in the terraform) and diskattachment without Alias parameter (when a customer didn't configure alias in the terraform).